src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -1,18 +1,654 @@
+import com.ctre.phoenix6.configs.Pigeon2Configurator;
+import com.ctre.phoenix6.hardware.Pigeon2;
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.commands.FollowPathCommand;
+import com.pathplanner.lib.config.RobotConfig;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+import com.pathplanner.lib.path.GoalEndState;
+import com.pathplanner.lib.path.IdealStartingState;
+import com.pathplanner.lib.path.PathConstraints;
 import com.pathplanner.lib.path.PathPlannerPath;
 import com.pathplanner.lib.path.Waypoint;
-
+import com.pathplanner.lib.util.DriveFeedforwards;
+import edu.wpi.first.math.MathUtil;
 import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.interpolation.InterpolatingDoubleTreeMap;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import frc.robot.utils.GeometryUtils;
+import frc.robot.utils.MatchStateUtil;
+import frc.robot.utils.PoseBuffer;
+import java.util.List;
+import java.util.Optional;
 
 public class DriveSubsystem extends SubsystemBase {
-    /* TODO: THIS SUBSYSTEM IS A PLACEHOLDER AND ONLY CONTAINS PLACEHOLDER METHODS */
-    public void driveToPoint(Pose2d point){
-        /* PLACEHOLDER METHOD */
+
+  private double targetHeadingDegrees;
+
+  double tagTargetHeading = 0.0;
+
+  // Create SwerveModules
+  public final SwerveModule frontLeft = new SwerveModule(
+      RobotMap.FRONT_LEFT_DRIVING_CAN_ID,
+      RobotMap.FRONT_LEFT_TURNING_CAN_ID,
+      DriveCal.FRONT_LEFT_CHASSIS_ANGULAR_OFFSET_RAD);
+
+  public final SwerveModule frontRight = new SwerveModule(
+      RobotMap.FRONT_RIGHT_DRIVING_CAN_ID,
+      RobotMap.FRONT_RIGHT_TURNING_CAN_ID,
+      DriveCal.FRONT_RIGHT_CHASSIS_ANGULAR_OFFSET_RAD);
+
+  public final SwerveModule rearLeft = new SwerveModule(
+      RobotMap.REAR_LEFT_DRIVING_CAN_ID,
+      RobotMap.REAR_LEFT_TURNING_CAN_ID,
+      DriveCal.BACK_LEFT_CHASSIS_ANGULAR_OFFSET_RAD);
```

Comment:
i think i said this earlier but why is this the only one to use "BACK" instead of "REAR?"